core/src/main/scala/mongo4cats/collection/queries/WatchQueryBuilder.scala

```diff
@@ -16,108 +16,153 @@
-    * @return
-    *   WatchQueryBuilder
-    */
-  def maxAwaitTime(maxAwaitTime: Duration): WatchQueryBuilder[F, T] =
-    WatchQueryBuilder(observable, WatchCommand.MaxAwaitTime[T](maxAwaitTime) :: commands)
-
-  /** Sets the logical starting point for the new change stream.
-    *
-    * @param resumeToken
-    *   the resume token
-    * @return
-    *   WatchQueryBuilder
-    */
-  def resumeAfter(resumeToken: BsonDocument): WatchQueryBuilder[F, T] =
-    WatchQueryBuilder(observable, WatchCommand.ResumeAfter[T](resumeToken) :: commands)
-
-  /** Similar to {@code resumeAfter}, this option takes a resume token and starts a new change stream returning the first notification after
-    * the token.
-    *
-    * <p>This will allow users to watch collections that have been dropped and recreated or newly renamed collections without missing any
-    * notifications.</p>
-    *
-    * <p>Note: The server will report an error if both {@code startAfter} and {@code resumeAfter} are specified.</p>
-    *
-    * @param startAfter
-    *   the startAfter resumeToken
-    * @return
-    *   WatchQueryBuilder
-    */
-  def startAfter(startAfter: BsonDocument): WatchQueryBuilder[F, T] =
-    WatchQueryBuilder(observable, WatchCommand.StartAfter[T](startAfter) :: commands)
-
-  /** The change stream will only provide changes that occurred after the specified timestamp.
-    *
-    * <p>Any command run against the server will return an operation time that can be used here.</p> <p>The default value is an operation
-    * time obtained from the server before the change stream was created.</p>
-    *
-    * @param startAtOperationTime
-    *   the start at operation time
-    * @since 1.9
-    * @return
-    *   WatchQueryBuilder
-    */
-  def startAtOperationTime(startAtOperationTime: BsonTimestamp): WatchQueryBuilder[F, T] =
-    WatchQueryBuilder(observable, WatchCommand.StartAtOperationTime[T](startAtOperationTime) :: commands)
-
-  def stream: fs2.Stream[F, ChangeStreamDocument[T]] =
-    applyCommands().boundedStream[F](1) // minimal latency
-
-  def boundedStream(capacity: Int): fs2.Stream[F, ChangeStreamDocument[T]] =
-    applyCommands().boundedStream[F](capacity)
+import scala.jdk.CollectionConverters._
+
+trait WatchQueryBuilder[F[_]] {
+  def batchSize(s: Int): WatchQueryBuilder[F]
+  def collation(c: model.Collation): WatchQueryBuilder[F]
+  def fullDocument(f: JFullDocument): WatchQueryBuilder[F]
+  def maxAwaitTime(d: Duration): WatchQueryBuilder[F]
+  def resumeAfter(rt: BsonDocument): WatchQueryBuilder[F]
+  def startAfter(sa: BsonDocument): WatchQueryBuilder[F]
+  def startAtOperationTime(saot: BsonTimestamp): WatchQueryBuilder[F]
+
+  //
+  def session(cs: ClientSession[F]): WatchQueryBuilder[F]
+  def noSession: WatchQueryBuilder[F]
+  def pipeline(p: Aggregate): WatchQueryBuilder[F]
+
+  //
+  def stream: Stream[F, ChangeStreamDocument[BsonValue]]
+  def boundedStream(c: Int): Stream[F, ChangeStreamDocument[BsonValue]]
```

Comment:
I think we should remove `boundedStream` from all builders, it isn't well defined and I think existed in the upstream impl because they didn't have any backpressure, as such it doesn't make much sense in this impl.

Reply:
:+1: